### PR TITLE
Add location endpoints

### LINKED ISSUE #11: Create endpoint for locations
# Overview

Create a django view that takes in a geometry (circle or rectangle) and returns a json list of those objects

# Context

Blocked by #8

### PR DESCRIPTION
**Pull request by travissouthard**
# Overview

Adds endpoints for the `location` object with list and detail views. Both of these are GET only since we users won't be adding their own locations. We can add them via the process developed during #22 or via the Django admin.

### Note

I couldn't get tags to work as part of the form and couldn't get hours to behave correctly. I think we will need to revisit our models and serializers to ensure they are what we need them to be

Closes #11 

# Demo

List view:
image

Detail view: 
image

# Testing instructions

- Create a super user
  - In one terminal tab run `docker-compose up --build` and leave that open
  - In a new terminal tab:
    - run 'docker exec -it third-places-django-1 bash`
    - Once in the django container's shell, run `python3 manage.py createsuperuser`
    - Follow the form to create a superuser with any name, password, etc. This is just for you
    - Exit the container shell with CTRL-D
- In your browser, go to [http://localhost:8321/admin/](http://localhost:8321/admin/)
  - Add 1 or 2 locations with the "Add" button for locations
- Navigate to [http://localhost:8321/api/locations](http://localhost:8321/api/locations/) and ensure:
  - [x] You see the locations you made
- Navigate to [http://localhost:8321/api/locations/1](http://localhost:8321/api/locations/1) and ensure:
  - [x] You see the location you made
- Navigate to [http://localhost:8321/api/locations/200](http://localhost:8321/api/locations/200) and ensure:
  - [x] You see a 404 response

# Checklist before merge

- [ ] Update the CHANGELOG.md
- [ ] Rebase your changes from whatever branch you're merging to

### CODE REVIEW

File: src/django/api/admin.py [hunk 1]
```diff
@@ -1,4 +1,15 @@
 from django.contrib import admin
-from .models import ThirdPlaceUser
+from .models import Location, ThirdPlaceUser
 
 admin.site.register(ThirdPlaceUser)
```
Comment (mdarkcity) [inline src/django/api/admin.py:4]: had to register the BusinessHours and LocationType models here:
```py
admin.site.register(BusinessHours)
admin.site.register(LocationType)
```
in order to add items to the corresponding tables and then select from these in the "Add location" screen. the `hours` field will need to be converted to an array of `BusinessHours` (ManyToManyField?) – or there could be a column for each day of the week (`mon_hours`, `tues_hours`, etc).

anyway, this works!

**Review by mdarkcity — APPROVED** [review]

**Review by zimurgh — APPROVED** [review]
Agreed with Max regarding BusinessHours needing to be many to many, wish I caught that when I setup the models. Perhaps we might be able to troubleshoot the tags field on Tuesday, hours is going to be the same story I imagine.

**Comment by travissouthard** [comment]
Thank you @zimurgh and @mdarkcity for your input! It was very helpful and this should be ready to test again. I adjusted the relations between tags, locations, and business hours to be more in line with what we intended to build. The instructions above should still apply. As you make a new location, there are plusses to add instances of the related objects as you make the location.

This works much better now.

**Review by mdarkcity — APPROVED** [review]
ManyToMany FTW!

**Pull request merged**